rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -787,6 +788,24 @@ void main_window::HandlePupInstallation(QString file_path)
 	Emu.SetForceBoot(true);
 	Emu.Stop();
 
+	struct fonts_manage
+	{
+		main_window* mw;
+
+		fonts_manage(main_window* _mw)
+			: mw(_mw)
+		{
+			// Remove possibly PS3 fonts from database
+			QFontDatabase::removeAllApplicationFonts();
+		}
+
+		~fonts_manage()
+		{
+			// Update with newly installed PS3 fonts
+			Q_EMIT mw->RequestGlobalStylesheetChange();
```

Comment:
just call this before creating the firmware cache. no need for some weird struct.

Reply:
It's for proper error handling, noone said installation always succeeds.

Reply:
it doesn't make any sense. there are at least 5 returns in this function before we even start installing the firmware

Reply:
Nvm, fixed.